JustSaying.Models/MessageAttributeValue.cs

```diff
@@ -0,0 +1,54 @@
+/* The code in this file is a subset of the MessageAttributeValue class as defined here:
+ * https://github.com/aws/aws-sdk-net/blob/0b18c061daf81f4966fddd3a3cbe953101394282/sdk/src/Services/SimpleNotificationService/Generated/Model/MessageAttributeValue.cs
+ */
+
+
+using System.IO;
+
+
+namespace JustSaying.Models
+{
+    /// <summary>
+    /// The user-specified message attribute value. For string data types, the value attribute
+    /// has the same restrictions on the content as the message body. For more information,
+    /// see <a href="http://docs.aws.amazon.com/sns/latest/api/API_Publish.html">Publish</a> in the AWS docs.
+    /// 
+    ///  
+    /// <para>
+    /// Name, type, and value must not be empty or null. In addition, the message body should
+    /// not be empty or null. All parts of the message attribute, including name, type, and
+    /// value, are included in the message size restriction, which is currently 256 KB (262,144
+    /// bytes). For more information, see <a href="http://docs.aws.amazon.com/sns/latest/dg/SNSMessageAttributes.html">Using
+    /// Amazon SNS Message Attributes</a> in the AWS docs.
+    /// </para>
+    /// </summary>
+    public class MessageAttributeValue
+    {
+        /// <summary>
+        /// Gets and sets the property BinaryValue. 
+        /// <para>
+        /// Binary type attributes can store any binary data, for example, compressed data, encrypted
+        /// data, or images.
+        /// </para>
+        /// </summary>
+        public MemoryStream BinaryValue { get; set; }

```

Comment:
Could _maybe_ make this a `Stream` so it's the least derived type, unless there's things that would consume this that are dependent on the fact it's a stream backed by memory rather than something else.

Reply:
Err, so in JustSaying we would have to convert the Stream into a memory stream for the AWS Implementation... presumably by reading the stream into a memory stream?

Reply:
Isn't this an output from AWS more than an input to AWS? If so, the reader just uses the stream, so the implementation of the stream doesn't matter.

Reply:
The binary value that this represents can only be up to 256 bytes long, so I think Stream might convey the wrong idea.

I think `MemoryStream` is a little weird here, and a `byte[]` would be the most idiomatic thing here (I know this was copy&pasta from AWS but doesn't feel like we have to keep it the same).